### ISSUE TITLE
"route: ['foo', 'bar', '{integers}']" doesn't match but "route: 'foo.bar[{integers}]'" does?!?

### ISSUE THREAD

**Issue by jhamlet** [Netflix/falcor-router#187]
@jhusain, @trxcllnt, @michaelbpaulson 

Ran in to some confusing behavior in defining routes today.

Apparently the _special_ path segments `'{integers}' '{ranges}'` and `'{keys}'` are only available in routes given as `Strings`.

Is this intended, or did I miss something with how to define routes as arrays?

This _does not_ work: `['foo', 'bar', '{integers}']` but this _does_ work `'foo.bar[{integers}]'`

---
Thanks,

;-j

**Comment by jhusain** [comment]
You're correct. We may provide a way to specify these with the Array syntax in the future. Before we do that though, we have to reserve some of the key space explicitly so that we do not conflict with keys included in the domain model by users. We're considering how to do this now.

**Comment by jhamlet** [comment]
Yeah. Figured that out. Once I switched to using strings everything worked.

The main reason I wanted to use arrays was to have access to the path as an array in the execution of the various get/set/call methods of the route object and slice/dice/julian it for returning path-values/jsonGraphs.

However, I came to realize (with judicious uses of `JSON.stringify`) that falcor-router is mutating my router's `route` property. The router replaces my string value with the equivalent array of `['foo', 'bar', { type: 'integers', named: false }, { type: 'keys', named: false }]` so I can still slice off the base part and use it my handlers.

Very un-intuitive (and shame on you for mutating my object), but the result is the same.

---

;-j
